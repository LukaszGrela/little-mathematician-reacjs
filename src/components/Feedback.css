.feedback {
    margin-top: -50px;
    height: 50px;
}
.feedback .mood-icon {
    display: inline-block;
    height: 32px;
    width: 32px;

    position: relative;
    top: 9px;

    margin-right: 8px;
}
.feedback.incorrect .mood-icon {
    fill: red;
}
.feedback.correct .mood-icon {
    fill: green;
}

.feedback button {
    display: inline-block;
    background-color: transparent;
    border: none;
    width: 40px;
    height: 40px;
    cursor: pointer;
    position: relative;
    top: 5px;
    margin-left: 8px;
}

.feedback button svg {
    -webkit-transition: all .5s ease;
    transition: all .5s ease;
}
.feedback button:hover svg {
    transform: rotate(1turn);
}
.feedback .message {
    display: inline-block;
}



.feedback .message .correct-answer {
    color: green;
    font-weight: bold;
    font-size: 1.25em;
    margin-left: 8px;
}