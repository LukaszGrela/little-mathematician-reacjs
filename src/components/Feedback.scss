/*
   Copyright 2018 ≈Åukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

@import '../sass/_variables.scss';
.feedback {
    margin-top: -57px;
    margin-bottom: 15px;
    height: 50px;
    
    .row {
        text-align: center;
        align-items: center;
        display: flex;
        margin: 0 auto;
        width: $width-game-column;
    }
    .cell {
        display: inline-block;
        flex-grow: 1;
    }
    .mood-icon {
        display: inline-block;
        height: 32px;
        width: 32px;
        /*
        position: relative;
        top: 9px;
        */
        padding: 4px;
        margin: 4px;
    }
    /* message */
    .message {
        display: inline-block;
        .correct-answer {
            color: $color-correct;
            font-weight: bold;
            font-size: 1.25em;
            margin-left: 8px;
        }
    }
    /* navigate button */
    button {
        display: inline-block;
        background-color: transparent;
        border: none;
        width: 40px;
        height: 40px;
        cursor: pointer;
        
        margin-left: 8px;
        padding: 0;
        svg {
            fill: $primary-text-color;
            -webkit-transition: all .5s ease;
            transition: all .5s ease;
            width: 38px;
            height: 38px;
            &:hover {
                transform: rotate(1turn);
            }
        }
    }
}

.feedback.incorrect .mood-icon {
    fill: $color-incorrect;
}

.feedback.correct .mood-icon {
    fill: $color-correct;
}