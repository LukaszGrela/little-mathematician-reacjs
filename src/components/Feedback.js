/*
   Copyright 2018 ≈Åukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/
import React, { Component } from 'react';
import MoodIcon from '../icons/MoodIcon';

import './Feedback.scss'
import ButtonIconNext from '../icons/ButtonIconNext';

class Feedback extends Component {

    promptAnimation = false;

    render() {
        const { answer, correct } = this.props;
        return (
            <div className={'feedback' + (answer.correct ? ' correct' : ' incorrect')}>
                <div className='row'>
                    <MoodIcon className='cell' isGood={answer.correct} />
                    <div className='message cell'>
                        {
                            answer.correct ?
                                'Well done!'
                                :
                                ['Not so, correct answer is ', <span key='correct-span' className='correct-answer'>{correct}</span>]
                        }
                    </div>
                    <button className={'cell' + (this.promptAnimation ? ' animate' : '')} onClick={() => {
                        this.props.onAction && this.props.onAction();
                    }}><ButtonIconNext /></button>
                </div>
            </div>
        );
    }
}

export default Feedback;
