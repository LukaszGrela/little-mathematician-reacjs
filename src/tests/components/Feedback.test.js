import React from 'react';
import { shallow } from 'enzyme';
import Feedback from '../../components/Feedback';

let wrapper, correct, answer, onAction;
beforeEach(() => {
    correct = 'A';
    answer = { correct: true };
    onAction = jest.fn();
    wrapper = shallow(<Feedback
        correct={correct}
        answer={answer}
        onAction={onAction} />);
})

test('Should render Feedback correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Should render Feedback correctly on wrong answer', () => {
    wrapper.setProps({ correct: 'B', answer: { correct: false } });
    expect(wrapper).toMatchSnapshot();
});

test('Should the Feedback display correct answer', () => {
    const value = 'B';
    wrapper.setProps({ correct: value, answer: { correct: false } });
    const span = wrapper.find('span.correct-answer');
    expect(span.length).toBe(1);
    expect(span.text()).toBe(value);
});

test('Should the Feedback display not good Mood', () => {
    const value = 'B';
    wrapper.setProps({ correct: value, answer: { correct: false } });
    const mood = wrapper.find('MoodIcon.cell');
    expect(mood.length).toBe(1);
    expect(mood.prop('isGood')).toBe(false);
});

test('Should the Feedback className have incorrect', () => {
    const value = 'B';
    wrapper.setProps({ correct: value, answer: { correct: false } });
    expect(wrapper.find('.feedback.incorrect')).toHaveLength(1);
    //and revert
    wrapper.setProps({ correct: value, answer: { correct: true } });
    expect(wrapper.find('.feedback.incorrect')).toHaveLength(0);
});

test('Should the button trigger onAction call', () => {
    const button = wrapper.find('button.cell');
    expect(button).toHaveLength(1);
    // simulate click
    button.simulate('click');
    expect(onAction).toBeCalled();
});