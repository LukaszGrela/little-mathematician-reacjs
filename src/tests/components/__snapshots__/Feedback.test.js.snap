// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Feedback correctly 1`] = `
<div
  className="feedback correct"
>
  <div
    className="row"
  >
    <MoodIcon
      className="cell"
      isGood={true}
    />
    <div
      className="message cell"
    >
      Well done!
    </div>
    <button
      className="cell animate"
      onClick={[Function]}
    >
      <ButtonIconNext />
    </button>
  </div>
</div>
`;

exports[`Should render Feedback correctly on wrong answer 1`] = `
<div
  className="feedback incorrect"
>
  <div
    className="row"
  >
    <MoodIcon
      className="cell"
      isGood={false}
    />
    <div
      className="message cell"
    >
      Not so, correct answer is 
      <span
        className="correct-answer"
        key="correct-span"
      >
        B
      </span>
    </div>
    <button
      className="cell animate"
      onClick={[Function]}
    >
      <ButtonIconNext />
    </button>
  </div>
</div>
`;
